### PR TITLE
#37/explore page

### LINKED ISSUE #37: `ExplorePage.tsx` 📡
### Expected Behavior

Style the `ExplorePage` component according to our figma

### Current Behavior

It's a dummy component from when navigation was set up

### Any other notes

We still need to know what `tags` go on here and this will probably need some changes made to the backend **

Image

### Assignment

This issue is free for anyone to take

### PR DESCRIPTION
**Pull request by nichgalzin**
# Description

**Closes #37**  
Explore page UI
- Adds html and styling for the explore page
- Adds store value exploreProjectId to allow for dynamic render of ExploreProject page
- implements card functionality to navigate to the explore project page and render correct data

### Files changed

- ```src/background/messages/fetchAllProjects.ts``` adds new listener for fetching all projects end point.
- ```src/components/FilterTags/FilterTags.tsx``` & ```src/components/FilterTags/FilterTags.css``` component for filter tags.
- ```src/components/ProjectCards/ProjectCard.css``` & src/components/ProjectCards/ProjectCard.tsx``` component for project cards.
- ```src/components/page-components/ExplorePage/ExplorePage.tsx``` & ```src/components/page-components/ExplorePage/ExplorePage.css``` updates explore html and styling.
- ```src/store/store.ts``` adds new value to store to control ExploreProjectPage id value.
- ```src/types/projectTypes.ts``` updates project types for new project response call.

### UI changes
Screenshot 2024-10-10 at 12 29 55

Screenshot 2024-10-10 at 12 30 15

# Tests

Include a description of any tests added - if none have been added explain why.

### CODE REVIEW

File: src/components/page-components/ExplorePage/ExplorePage.css [hunk 1]
```diff
@@ -1,5 +1,5 @@
 .explore-page {
-  background-color: red;
+  background-color: white;
```
Comment (camelPhonso) [inline src/components/page-components/ExplorePage/ExplorePage.css]: I know this one seems picky because it's just a white background but that should be `var(--base)` in case their colour scheme drastically changes in future (or dark mode is introduced or whatever)

File: src/background/messages/fetchAllProjects.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+import { allProjectQueryString } from "~queries/allProjectsQuery"
+import type { CurrentProjectResponse } from "~types/projectTypes"
+import { fetchStrapiContent } from "~utils/fetchStrapiContent"
+
+const handler: PlasmoMessaging.MessageHandler = async (req, res) => {
+  const response = await fetchStrapiContent<CurrentProjectResponse>(
+    `api/projects?${allProjectQueryString}`
+  )
+  res.send(response)
+}
```
Comment (camelPhonso) [inline src/background/messages/fetchAllProjects.ts:12]: This is all fine but maybe we should add an issue to the board about pagination for this call?

I think this is fine to merge as is now but I can already see this page loading a bit slowly and so we should maybe think about introducing pages to this api endpoint and keeping the number of projects being loaded down to a sensible max number

File: src/components/ProjectCards/ProjectCard.css [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  width: 165px;
+  height: 165px;
+  border-radius: 0.438rem;
+  object-fit: cover;
+}
+
+.project-card img:hover {
+  scale: 0.97;
+  transition: scale 100ms ease;
+}
```
Comment (camelPhonso) [inline src/components/ProjectCards/ProjectCard.css:16]: also not something we need to sort now but I think this might be a departure from the design we have - maybe if Rich is coming back to review things we should prepare a list of changes we want reviewed and have this on there.

I rather think this would probably be a change in the drop-shadow instead of a scaling animation to be in line with their Y2K-y0lo aesthetic

File: src/components/FilterTags/FilterTags.tsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import "./FilterTags.css"
+
+const FilterTags = () => {
+  return (
+    <div className="flex tags-container">
+      {tags.map(tag => {
+        return (
+          <button
+            key={tag.id}
+            className="filter-button"
```
Comment (camelPhonso) [inline src/components/FilterTags/FilterTags.tsx]: `button--filter` please for consistency ?

**Review by camelPhonso — APPROVED** [review]
